gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/manager/impl/ApiManagerImpl.java

```diff
@@ -76,7 +83,23 @@ private boolean register(ReactableApi<?> api, boolean force) {
         // Get deployed API
         ReactableApi<?> deployedApi = get(api.getId());
 
-        // Does the API have a matching sharding tags ?
+        List<Plugin> plugins;
+        if (api.getDefinitionVersion() == DefinitionVersion.V4) {
```

Comment:
Does this block exist in the 🧅 architecture? If so, could we use it or are we obliged to rewrite it here?

Reply:
No onions here. 👍

Reply:
This APIManager is in the gateway, so it doesn't access the 🧅 architecture services. This si why we have to repeat this check here